### ISSUE TITLE
[BUG] Content->getImage() method breaks for images with alt and no path

### ISSUE THREAD

**Issue by ToBe998** [bolt/bolt#5900]
The function `Content->getImage()` takes the first field of type *image* and the returns either the image path if we have a *file* key or the whole thing if not (as fallback for bolt 1.5.0).

If a user now edits the alt title of an image but does not provide a image path, this check fails and returns the array [alt: „the alt“]. 
Since every part of bolt (including especially the twig template of the backend) assume it’s a string and not an array, this throws an exception, making some views of the backend unusable. 

**The method in question** - ContentValuesTrait.php Line 503
```
        // Grab the first field of type 'image', and return that.
        foreach ($this->contenttype['fields'] as $key => $field) {
            if ($field['type'] == 'image') {
                // After v1.5.1 we store image data as an array
                if (is_array($this->values[$key]) && isset($this->values[$key]['file'])) {
                    return $this->values[$key]['file'];
                }

                return $this->values[$key];
            }
        }
```

Example of a breaking template with a "Array to String Conversion":
*_base/_listing.twig*
```
    {% set thumb_title = __('Image') ~ ': ' ~ content.getImage %}
```

In my case a widget that uses the @bolt/_base/listing.twig block creating the thumbnails breaks, making the dashboard fail critically (Remember: we can't catch exceptions in twig).

**Proposed Fix** - ContentValuesTrait.php Line 503
```
        // Grab the first field of type 'image', and return that.
        foreach ($this->contenttype['fields'] as $key => $field) {
            if ($field['type'] == 'image') {
                // After v1.5.1 we store image data as an array
                if (is_array($this->values[$key])) {
                    return isset($this->values[$key]['file']) ? $this->values[$key]['file'] : "";
                }

                return $this->values[$key];
            }
        }
```

Details
-------

 - Relevant Bolt Version: 3.0.12
 - Install type: Composer install
 - PHP version: 5.5
 - Used web server: Apache
 - For UX/UI issues: Chrome

Reproduction
------------
- Create a content type with a field of type *image* with alt attribute
- Enter an alt value and save
- Use the $record->getImage() method somewhere and observe returned file path

**Comment by GawainLynch** [comment]
How tired am I … I think I just spent 10 minutes staring at the code trying to figure it out too!

But yeah, there was an intention to fix the getter for 3.0 … buuuuuuuuuut. :wink: 

Want to PR it to get the glory?

**Comment by ToBe998** [comment]
Can do. I'll try to find the time tomorrow. What branch? master?
(I'd need it in 3.0 or maybe at least 3.1)

**Comment by GawainLynch** [comment]
3.0 is closed and locked, but 3.1 will be open until (at this stage) the 24th … so aim at `release/3.1` :+1:

**Comment by CarsonF** [comment]
Can we deprecate this method?

**Comment by ToBe998** [comment]
The method itself is still used widely. Just that fallback for 1.5.1 is probably very outdated.

**Comment by CarsonF** [comment]
I didn't say remove it. I said deprecate it. Doesn't it make more sense to reference the field name directly?

**Comment by ToBe998** [comment]
This method uses the first image it finds from any field in the contenttype. Similar to getExcerpt for texts. So it's still a feature that's not available otherwise, afaik.

**Comment by CarsonF** [comment]
But what does that allow?

**Comment by ToBe998** [comment]
Well, as i said, you can write very generic templates that display an image for a record, in whatever field it may be stored. Take the Bolt-Dashboard for example. It shows a list of all recent records of all types. Ad each item in the lists has an image, independant on how you setup your contenttypes.

Not usefull for good specific themes, but very usefull for generic stuff like the bolt backend.

**Comment by CarsonF** [comment]
Gotcha

**Comment by GawainLynch** [comment]
@CarsonF — the whole trait is, the docblock says everything, but the person that split it out of `\Bolt\Storage` apparently didn't put an `@deprecated` tag on it —> :koala: 

The reason it is still there is that we needed transitional elements we could use while the storage refactor continued, and 3.0-dev went on too long … and …

![](http://www.toonbarn.com/wordpress/wp-content/uploads/2009/12/Family-Guy-Something-Something-Something-Dark-Side.jpg)

**Comment by ToBe998** [comment]
Ah, you're probably speaking about the new Content object. I was referring to the old legacy content class, which still is used by the backend. Right?

**Comment by GawainLynch** [comment]
> Ah, you're probably speaking about the new Content object. 

Correct :+1: 

> I was referring to the old legacy content class, which still is used by the backend. Right?

Only on the dashboard, and only until 3.3 — if I can get that done next week, if not then 3.4 with the data object overhaul.

**Comment by ToBe998** [comment]
Ok, I'll try to find the respective lines in the new class too.

**Comment by GawainLynch** [comment]
Fixed in #5919
